### PR TITLE
Fix italy_yaml for the removal of SmartMappedCollection

### PR DESCRIPTION
**Pull request by veluca93**
The removal of SMC left some bugs in the italy_yaml loader, which this PR fixes.

---
This change is [Reviewable](https://reviewable.io/reviews/cms-dev/cms/744)

### CODE REVIEW

File: cmscontrib/loaders/italy_yaml.py [hunk 1]
```diff
@@ -662,6 +662,8 @@ def get_task(self, get_statement=True):
         elif public_testcases != "":
             for x in public_testcases.split(","):
                 args["testcases"][int(x.strip())].public = True
+        args["testcases"] = dict((tc.codename, tc) for tc in args["testcases"])
+        args["managers"] = dict((tc.filename, mg) for mg in args["managers"])
```
Comment (lerks) [inline cmscontrib/loaders/italy_yaml.py]: `mg.filename`

**Comment by stefano-maggiolo** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

*[cmscontrib/loaders/italy_yaml.py, line 366 at r1](https://reviewable.io:443/reviews/cms-dev/cms/744#-Kh5l6VPz0GvE1xEDOuh:-Kh5l6VQzxUnQSJPAdnn:b-s5ksv1) ([raw file](https://github.com/cms-dev/cms/blob/cc717d76bf9a4c9229a52772a6dc90ddc0153d63/cmscontrib/loaders/italy_yaml.py#L366)):*
> ```Python
>                 logger.critical("Couldn't find any task statement, aborting.")
>                 sys.exit(1)
>             args["statements"] = [Statement(primary_language, digest)]
> ```

This needs to be changed too:
```py
            args["statements"] = {
                primary_language: Statement(primary_language, digest)
            }
```

---

*[cmscontrib/loaders/italy_yaml.py, line 656 at r1](https://reviewable.io:443/reviews/cms-dev/cms/744#-Kh5lvMzTwtMdhC6Wqa5:-Kh5lvMzTwtMdhC6Wqa6:blnamjk) ([raw file](https://github.com/cms-dev/cms/blob/cc717d76bf9a4c9229a52772a6dc90ddc0153d63/cmscontrib/loaders/italy_yaml.py#L656)):*
> ```Python
>             if args["task_type"] == "OutputOnly":
>                 task.attachments += [
>                     Attachment("input_%03d.txt" % i, input_digest)]
> ```

```py
                task.attachments.set(
                    Attachment("input_%03d.txt" % i, input_digest))
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cms-dev/cms/744)*

**Comment by veluca93** [comment]
I was probably half-asleep when I made those changes :)

**Comment by stefano-maggiolo** [comment]
:lgtm: @lerks, can you push after your other PR?

Also, the same probably applies to polygon, we should find a test task for it and add both to our tests.

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cms-dev/cms/744#-:-Kh8Nd8mGs1CnS19XA_O:b-ajjir0)*

**Comment by lerks** [comment]
Pushed as 77dd16692d51805fa5652fb69b4a23c6cf821a28.

**Pull request closed without merging**